src/vm/i386/cgenx86.cpp

```diff
@@ -572,29 +594,30 @@ void FaultingExceptionFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     pRD->PCTAddr = GetReturnAddressPtr();
 
-#ifndef WIN64EXCEPTIONS
-    CalleeSavedRegisters* regs = GetCalleeSavedRegisters();
+#ifdef WIN64EXCEPTIONS
 
-#define CALLEE_SAVED_REGISTER(regname) pRD->p##regname = (DWORD*) &regs->regname;
-    ENUM_CALLEE_SAVED_REGISTERS();
-#undef CALLEE_SAVED_REGISTER
+    memcpy(pRD->pCurrentContext, &m_ctx, sizeof(CONTEXT));
 
-    pRD->SP = m_Esp;
-    pRD->ControlPC = *PTR_PCODE(pRD->PCTAddr);
+    pRD->SP = m_ctx.Esp;
+    pRD->ControlPC = m_ctx.Eip;
 
-#else // WIN64EXCEPTIONS
+#define CALLEE_SAVED_REGISTER(regname) pRD->pCurrentContextPointers->regname = &m_ctx.regname;
```

Comment:
Looking at the amd64 version, we should also set the context pointers for caller saved registers here.